.github/workflows/build.yaml

```diff
@@ -9,6 +9,76 @@ on:
     branches: [ master ]
 
 jobs:
+  changes:
+    name: changed files
+    runs-on: ubuntu-latest
+    # Set job outputs to values from changed_files step for other jobs
+    outputs:
+      codecheck: ${{ steps.changed_files.outputs.codecheck }}
+      clang_tidy: ${{ steps.changed_files.outputs.clang_tidy }}
+      run_test_suite: ${{ steps.changed_files.outputs.test_suite }}
+      inst_windows_msvc: ${{ steps.changed_files.outputs.inst_windows_msvc }}
+      inst_windows: ${{ steps.changed_files.outputs.inst_windows }}
+      inst_macos: ${{ steps.changed_files.outputs.inst_macos }}
+      appimage: ${{ steps.changed_files.outputs.appimage }}
+    steps:
+    - name: Check for changed files
+      uses: dorny/paths-filter@v2
+      id: changed_files
+      with:
+        # groups to check for changed files (as string, yaml anchors are resolved by the tool)
+        filters: |
+          codecheck:
+            - 'CMakeLists.txt'
+            - '*/CMakeLists.txt'
+            - 'cmake/codecheck/**'
+            - 'cmake/WlFunctions.cmake'
+            - 'src/**'
+            - '.github/scripts/install_deps.sh' # calls install-dependencies.sh
```

Comment:
All this should be just `*src_cmake_dependencies`.

Reply:
I kept it separate because it only needs part of cmake. But I agree, it is not a big difference.

Reply:
I haven't looked much into how codecheck works, but it's another C++ source analyser, so it may be affected by anything that changes how the code is built. (well, OK, `cmake` also handles linking, but I don't think we can separate those parts)

Reply:
codechecker is a python script.
The codechecker code only runs `cd build/; cmake .` (or `cmake build/`). For compiling, linking, ... it needs another command.
But let us use `*src_cmake_dependencies` for now. We might discuss this in an other pr (it is prepared).